src/node/rpc/frontend.h

```diff
@@ -273,6 +306,13 @@ namespace ccf
         return;
       }
 
+      // Before attempting to forward, make sure we're in the same View as we
+      // previously thought we were.
+      if (!check_session_consistency(ctx))
```

Comment:
There's a subtle hole here - we check the view we're executing in now, and then the primary we're forwarding to below. Technically there could be an election between these steps, and we'd forward our request to the wrong primary. This is fine - that primary will report a consistency break and tell us to close the session, rather than us doing so ourselves. But I wonder if, long-term, all of this consensus state should be snapped atomically once-per-transaction, rather than allowing it to vary during transaction execution.